app/controllers/packages_controller.rb

```diff
@@ -245,4 +257,15 @@ def validate_file_ids(file_ids, current_user, errors)
     return data_files.uniq
   end
 
+  def find_invalid_content_message(errors)
```

Comment:
I think it would be much easier to create a warnings array, and pass this into the validation method and add a message if need be. It would avoid having to do this messy find.